### DISCUSSION TITLE
Blog 2 - 20.10.2024

### DISCUSSION THREAD

**Discussion by JustMischka** [Anastylos/Jump-and-Run#35] [Blog]
# Blog 2 - Softwareanforderungsspezifikation

## 1. Einleitung
In diesem Blog werden die Anforderungen für Jump and Run Spiel spezifiziert. Die Softwareanforderungsspezifikation (SRS) dient als Grundlage für die weitere Entwicklung und definiert die erwarteten Funktionalitäten und Qualitätsmerkmale des Systems.

### 2.1 Benutzeranforderungen
- Die Figur bewegt sich nach links oder rechts, springt, hechtet und kann Pfeile schießen
- Gegner verursachen einen Lebensverlust, wenn sie die Figur treffen.
- Gegner können durch einen Sprung auf den Kopf oder durch Treffer mit Pfeilen besiegt werden.
- Fallen führen zum Verlust eines Lebens.
- Power-Ups erscheinen zufällig und haben zeitlich begrenzte Effekte.

### 2.2 Systemanforderungen
- Der Spielstand wird am Ende jedes Levels automatisch gespeichert.
- Der Spieler kann den Fortschritt laden und weiterspielen.
- Am Ende jedes Levels wird der Punktestand angezeigt und gespeichert.
- Die Punktezahl wird in einem Highscore gespeichert, welche im Hauptmenü oder im Pause-Screen angeschaut werden kann.

### 3.1 Leistungsanforderungen
- Ladezeiten zwischen Levels betragen maximal 3 Sekunden.

### 3.2 Sicherheitsanforderungen
- Spielstände werden lokal und in der Cloud gespeichert.
- Es gibt keine sensiblen Daten erfordernden In-Game-Käufe.

### 3.3 Zuverlässigkeitsanforderungen
- Die Steuerung ist intuitiv und benutzerfreundlich.
- Der Code ist modular und gut dokumentiert.
- Neue Levels, Power-Ups und Gegner können leicht hinzugefügt werden.

#### Hauptmenü:
Das Hauptmenü des 2D-Spiels "Jump and Run" ist in einem stilvollen Pixel-Art-Design gestaltet. Der Hintergrund zeigt eine malerische Waldszene mit hohen Bäumen und nebligen Bergen. Das Menü selbst bietet dem Spieler drei wesentliche Elemente: einen "Start Game" Button zum Spielstart, einen "Settings" Button für die Einstellungen sowie eine Highscore-Anzeige zur Aufzeichnung der Bestleistungen. Die gesamte Gestaltung in Grün- und Blautönen vermittelt eine natürliche, einladende Atmosphäre.

<img src="https://github.com/user-attachments/assets/79d27cae-9c2d-4d8b-bbe2-ab560d306467" alt="Hauptmenü Mockup" title="Hauptmenü des Jump and Run Spiels">

#### Spiel Aussschnitt:
Das Mockup zeigt einen Levelabschnitt des 2D-Plattformspiels im verspielten Pixel-Art-Stil. Der Spieler steuert einen kleinen weißen Charakter, der sich in einer bunten Naturlandschaft bewegt. Die Szene ist von einem leuchtend blauen Himmel überspannt und zeigt saftig grüne Bäume im Mittelgrund. Schwebende Plattformen, Hindernisse und einfache Gegner sind im Level verteilt.

<img src="https://github.com/user-attachments/assets/83dcc0fc-d16b-447f-a8ed-ffd49761ccd9" alt="Spiel Ausschnitt Mockup" title="Levelabschnitt des Jump and Run Spiels">

---
*Letzte Aktualisierung: 20.10.2024*

**Comment by schmid-di** [comment]
Das Konzept für das Jump-and-Run-Spiel ist klar und gut strukturiert. Die funktionalen Anforderungen wie Bewegungen, Gegnerinteraktionen und Power-Ups schaffen eine solide Grundlage für klassisches Gameplay. Besonders gefallen mir die nicht-funktionalen Anforderungen, z.B. kurze Ladezeiten und intuitive Steuerung, die das Spielerlebnis verbessern. Die Mockups im Pixel-Art-Stil vermitteln eine ansprechende, einladende Atmosphäre. Insgesamt ein durchdachtes Konzept mit Potenzial für ein unterhaltsames Spielerlebnis!

Euer UMOC Team
/JMS /PM

**Comment by SidMllr** [comment]
Die funktionalen Anforderungen in diesem Blogpost haben uns wirklich beeindruckt! Wir sind begeistert, wie präzise und durchdacht euer Team sie ausgearbeitet hat. Die dazugehörigen UML-Diagramme würden uns auch sehr interessieren. Die nicht-funktionalen Anforderungen sind klar und nachvollziehbar, erfahrungsgemäß werden diese bei Videospielen oft vernachlässigt, daher sind wir sehr froh zu sehen, dass euer Team einen Fokus auf beispielsweise die Ladezeiten legt.
Das UI-Mockup zeigt eure Vision für das Jump-and-Run-Spiel sehr gut und vermittelt einen klaren Eindruck von der Benutzeroberfläche. Das Pixel-Art-Design gefällt uns sehr!

Wir sind gespannt auf die weitere Entwicklung des Spiels und freuen uns darauf, es bald selbst auszuprobieren.

Freundliche Grüße  
Project HÆLTH
